src/m365/planner/commands/task/task-add.ts

```diff
@@ -0,0 +1,387 @@
+  bucketName?: string;
+  title: string;
+  startDateTime?: string;
+  dueDateTime?: string;
+  percentComplete?: number;
+  assignedToUserIds?: string;
+  assignedToUserNames?: string;
+  description?: string;
+  orderHint?: string;
+}
+
+class PlannerTaskAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.TASK_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner Task';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    telemetryProps.bucketId = typeof args.options.bucketId !== 'undefined';
+    telemetryProps.bucketName = typeof args.options.bucketName !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.startDateTime = typeof args.options.startDateTime !== 'undefined';
+    telemetryProps.dueDateTime = typeof args.options.dueDateTime !== 'undefined';
+    telemetryProps.percentComplete = typeof args.options.percentComplete !== 'undefined';
+    telemetryProps.assignedToUserIds = typeof args.options.assignedToUserIds !== 'undefined';
+    telemetryProps.assignedToUserNames = typeof args.options.assignedToUserNames !== 'undefined';
+    telemetryProps.description = typeof args.options.description !== 'undefined';
+    telemetryProps.orderHint = typeof args.options.orderHint !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['title', 'planId', 'bucketId'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getPlanId(args)
+      .then((planId: string) => 
+        this.getBucketId(args, planId).then(async (bucketId: string) => {
+          const assignments: { [userId: string]: any; } = await this.generateUserAssignments(args, logger);
+
+          const requestOptions: any = {
+            url: `${this.resource}/v1.0/planner/tasks`,
+            headers: {
+              'accept': 'application/json;odata.metadata=none'
+            },
+            responseType: 'json',
+            data: {
+              planId: planId,
+              bucketId: bucketId,
+              title: args.options.title,
+              startDateTime: args.options.startDateTime,
+              dueDateTime: args.options.dueDateTime,
+              percentComplete: args.options.percentComplete,
+              assignments: assignments,
+              orderHint: args.options.orderHint
+            }
+          };
+
+          return request
+            .post(requestOptions)
+            .then(async (newTask: any) => {
```

Comment:
Since we know the type of data that we'll receive from the API, let's not use `any` but rather the exact type so that we can benefit from design-time type safety features of TypeScript